SofaKernel/modules/Sofa.Geometry/src/sofa/geometry/Triangle.h

```diff
@@ -90,8 +90,120 @@ struct Triangle
         constexpr Node n{};
         static_assert(std::distance(std::begin(n), std::end(n)) == 3, "Triangle normal can only be computed in 3 dimensions.");
 
-        return (n1 - n0).cross(n2 - n0);
+        // Vec gives access to cross() and operator-
+        if constexpr (std::is_same_v < Node, sofa::type::Vec<3, T> >)
+        {
+            const auto a = n1 - n0;
+            const auto b = n2 - n0;
+
+            return a.cross(b);
+        }
+        else
```

Comment:
What are the other cases?

Reply:
using this fonction with somewhing than Vec (std::array or type::fixed_array for example)